### ISSUE TITLE
daemon: remove pipeline error if come jobs are related to it.

### ISSUE THREAD

**Issue by FeelyChau** [alibaba/pipcook#536]
```shell
$ pipcook pipeline remove 34fc5mef
✖ SQLITE_CONSTRAINT: FOREIGN KEY constraint failed
```

**Comment by yorkie** [comment]
How to reproduce the bug?

**Comment by FeelyChau** [comment]
> How to reproduce the bug?

Start a job from pipeline, then remove this pipeline.

**Comment by yorkie** [comment]
Alternatives:

1. remove all the jobs of the pipeline with prompt
2. mark the pipeline removed instead of removing it from table.

**Comment by rickycao-qy** [comment]
I will vote for 1. If I'm going to remove pipeline, I would like to remove all pipelines, jobs and all files (models) related to this pipeline. Just to prompt to notify users the consequence of removing is fine

**Comment by liangzr** [comment]
+1 for remove all jobs, once the pipeline removed by the user manually, jobs should release resources usage
> 1. remove all the jobs of the pipeline with prompt

**Comment by FeelyChau** [comment]
> +1 for remove all jobs, once the pipeline removed by the user manually, jobs should release resources usage
> 
> > 1. remove all the jobs of the pipeline with prompt

+1

**Comment by yorkie** [comment]
Ok, it seems 1st is better.

**Comment by FeelyChau** [comment]
Fixed by #547 .
